tool/tbot/config/configtemplate_tls.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"context"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/client/identityfile"
+	"github.com/gravitational/teleport/tool/tbot/identity"
+	"github.com/gravitational/trace"
+)
+
+const defaultTLSPrefix = "tls"
+
+// TemplateTLS is a config template that wraps identityfile's TLS writer.
+// It's not generally needed but can be used to write out TLS certificates with
+// alternative prefix and file extensions if needed for application
+// compatibility reasons.
+type TemplateTLS struct {
+	Prefix string `yaml:"prefix,omitempty"`
+}
+
+func (t *TemplateTLS) CheckAndSetDefaults() error {
+	if t.Prefix == "" {
+		t.Prefix = defaultTLSPrefix
+	}
+
+	return nil
+}
+
+func (t *TemplateTLS) Name() string {
+	return TemplateTLSName
+}
+
+func (t *TemplateTLS) Describe() []FileDescription {
+	return []FileDescription{
+		{
+			Name: t.Prefix + ".key",
+		},
+		{
+			Name: t.Prefix + ".crt",
+		},
+		{
+			Name: t.Prefix + ".cas",
+		},
+	}
+}
+
+func (t *TemplateTLS) Render(ctx context.Context, authClient auth.ClientI, currentIdentity *identity.Identity, destination *DestinationConfig) error {
+	dest, err := destination.GetDestination()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	hostCAs, err := authClient.GetCertAuthorities(ctx, types.HostCA, false)
```

Comment:
Databases should use Database CA here on Teleport 10+.

Reply:
Hmmm. This particular template is for general use so I think we'll need to make the CA configurable. The database-specific ones can set that by default, though.

I sort of wonder if we should make specific templates for MySQL / MariaDB / Postgres / etc just to mitigate any confusion.

Reply:
This is what I meant. To create a separate implementation for databases as they are using a different CA. Sorry for the confusion.

Reply:
No worries! For now I've just made this configurable:

```yaml
configs:
  - tls:
      ca_cert_type: database
```

I suspect we'll be revisiting database-specific configs in the future anyway, so dedicated database-specific configs should be okay as a follow-up PR.